target_chains/ethereum/sdk/js/src/examples/EvmBenchmark.ts

```diff
@@ -0,0 +1,117 @@
+import Web3 from "web3";
+import yargs from "yargs";
+import { hideBin } from "yargs/helpers";
+
+import { EvmPriceServiceConnection } from "../index";
+import HDWalletProvider from "@truffle/hdwallet-provider";
+import PythInterfaceAbi from "@pythnetwork/pyth-sdk-solidity/abis/IPyth.json";
+
+const argv = yargs(hideBin(process.argv))
+  .option("network", {
+    description: "RPC of the network to relay on.",
+    type: "string",
+    required: true,
+  })
+  .option("endpoint", {
+    description:
+      "Endpoint URL for the price service. e.g: https://endpoint/example",
+    type: "string",
+    required: true,
+  })
+  .option("pyth-contract", {
+    description: "Pyth contract address.",
+    type: "string",
+    required: true,
+  })
+  .option("price-id", {
+    description:
+      "Price feed id (in hex) to fetch" + " e.g: 0xf9c0172ba10dfa4d19088d...",
+    type: "string",
+    required: true,
+  })
+  .option("timestamp", {
+    description: "Timestamp of the prices to fetch" + " e.g., 2022-", // TODO
+    type: "string",
+    required: true,
+  })
+  .option("mnemonic", {
+    description: "Mnemonic (private key) for sender",
+    type: "string",
+    required: true,
+  })
+  .help()
+  .alias("help", "h")
+  .parserConfiguration({
+    "parse-numbers": false,
+  })
+  .parseSync();
+
+const network = argv.network;
+const pythContractAddr = argv.pythContract;
+
+const connection = new EvmPriceServiceConnection(argv.endpoint);
+
+async function run() {
+  const provider = new HDWalletProvider({
+    mnemonic: {
+      phrase: argv.mnemonic,
+    },
+    providerOrUrl: network,
+  });
+
+  // @ts-ignore
+  const web3 = new Web3(provider);
+  const priceId = argv.priceId as string;
+  // The unix timestamp in seconds
+  const unixTimestamp = Date.parse(argv.timestamp) / 1000;
+
+  console.log(`Querying unix timestamp: ${unixTimestamp}`);
+
+  const [priceFeedUpdateVaa, updateTimestamp] = await connection.getVaa(
+    priceId,
+    unixTimestamp
+  );
+  console.log(`Next pyth update was at: ${updateTimestamp}`);
+  console.log(priceFeedUpdateVaa);
+
+  const priceFeedUpdate =
+    "0x" + Buffer.from(priceFeedUpdateVaa, "base64").toString("hex");
```

Comment:
it's kind of janky that the user has to convert the data format. We should fix this as part of our overall client js refactoring.